### ISSUE TITLE
About training loss function 

### ISSUE THREAD

**Issue by hek14** [aritra0593/Reinforced-Feature-Points#1]
Thank you for you excellent work! The paper really gives me a lot of inspiration. I have a question about the training loss function. In your paper, the (6) equation says: 
![Screen Shot 2020-07-09 at 3 07 47 PM](https://user-images.githubusercontent.com/22982681/87008349-2da31700-c1f6-11ea-8b86-30034bb7811b.png)

Do you implement a gradient backward function or just use something like:  
![image](https://user-images.githubusercontent.com/22982681/87009136-5546af00-c1f7-11ea-9468-86733157498c.png)

**Comment by aritra0593** [comment]
Hi, the loss function does not have log prob. It comes in the differentiation equation only.
Now, coming to your question, can you please elaborate on which gradients you are talking about?

**Comment by hek14** [comment]
Thank you for your reply! Sorry for the late response. I'm just wondering about the gradients backward process, you said that above loss function comes in differentiation equation only, I write some pseudocode: 
![Screen Shot 2020-07-16 at 9 00 19 PM](https://user-images.githubusercontent.com/22982681/87673587-5eabba80-c7a7-11ea-9a7f-781f50422d45.png)

Do I understand correctly?

**Comment by JLtwoP** [comment]
do you reproduce the training pipeline? I dont understand about the training loss function

**Comment by aritra0593** [comment]
Hi, we will be releasing it by the end of this month

**Comment by aritra0593** [comment]
Hi, I have added the training pipeline. I will be commenting parts of the training files in the days to come. Currently it prints in each epoch, and I would like to fix some printing statements. But the whole end to end training pipeline works with RANSAC now. I have also added the training datasets.

Let me know if you have some questions about the codes.
